roofit/roofitcore/src/TestStatistics/LikelihoodWrapper.cxx

```diff
@@ -0,0 +1,159 @@
+{
+   return likelihood_->defaultErrorLevel();
+}
+std::string LikelihoodWrapper::GetName() const
+{
+   return likelihood_->GetName();
+}
+std::string LikelihoodWrapper::GetTitle() const
+{
+   return likelihood_->GetTitle();
+}
+
+void LikelihoodWrapper::enableOffsetting(bool flag)
+{
+   do_offset_ = flag;
+   // Clear offset if feature is disabled so that it is recalculated next time it is enabled
+   if (!do_offset_) {
+      offset_ = 0;
+      offset_carry_ = 0;
+   }
+}
+
+void LikelihoodWrapper::setOffsettingMode(OffsettingMode mode)
+{
+   offsetting_mode_ = mode;
+   if (isOffsetting()) {
+      oocoutI(static_cast<RooAbsArg *>(nullptr), Minimization) << "LikelihoodWrapper::setOffsettingMode(" << GetName() << "): changed offsetting mode while offsetting was enabled; resetting offset values" << std::endl;
+      offset_ = 0;
+      offset_carry_ = 0;
+   }
+}
+
+void LikelihoodWrapper::applyOffsetting(double &current_value, double &carry)
+{
+   if (do_offset_) {
+
+      // If no offset is stored enable this feature now
+      if (offset_ == 0 && current_value != 0) {
+         offset_ = current_value;
+         offset_carry_ = carry;
+         if (offsetting_mode_ == OffsettingMode::legacy) {
+            auto sum_likelihood = dynamic_cast<RooSumL*>(likelihood_.get());
+            if (sum_likelihood != nullptr) {
+               double subsidiary_value, subsidiary_carry;
+               std::tie(subsidiary_value, subsidiary_carry) = sum_likelihood->getSubsidiaryValue();
+               // "undo" the addition of the subsidiary value to emulate legacy behavior
+               offset_ -= subsidiary_value;
+               offset_carry_ -= subsidiary_carry;
+               // then add 0 in Kahan summation way to make sure the carry gets taken up into the value if it should be
+               double y = 0 - offset_carry_;
+               double t = offset_ + y;
+               offset_carry_ = (t - offset_) - y;
+               offset_ = t;
+               // also set carry to this value, again to emulate legacy behavior
+               carry = offset_carry_;
+            }
+         }
+         oocoutI(static_cast<RooAbsArg *>(nullptr), Minimization)
+            << "LikelihoodWrapper::applyOffsetting(" << GetName() << "): Likelihood offset now set to " << offset_
+            << std::endl;
+      }
+
+      // Subtract offset
+      // old method:
+//      {
+//         double y = -offset_ - (carry + offset_carry_);
+//         double t = current_value + y;
+//         carry = (t - current_value) - y;
+//         current_value = t;
+//      }
+      // Jonas method:
+      {
+         double new_value = current_value - offset_;
+         double new_carry = carry - offset_carry_;
+         // then add 0 in Kahan summation way to make sure the carry gets taken up into the value if it should be
+         double y = 0 - new_carry;
+         double t = new_value + y;
+         carry = (t - new_value) - y;
+         current_value = t;
+      }
```

Comment:
Can probably be replaced by `KahanSum`

Reply:
Yeah, this is where I was quickly trying out the method to see whether it made a difference in the results. Will replace.

Reply:
Added TODO, should check results after replacement.